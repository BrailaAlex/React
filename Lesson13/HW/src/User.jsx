import React from "react";

class User extends React.Component {
  state = {
    avatarUrl: null,
    name: null,
    location: null,
  }

  componentDidMount() {
    debugger;
    this.fetchUserInfo(this.props.match.params.userId)
  }

  componentDidUpdate(prevProp) {
    debugger;
    if (this.props.match.params.userId !== prevProp.match.params.userId) {
      this.fetchUserInfo(this.props.match.params.userId);
    }
  }

  fetchUserInfo = (userData) => {
    debugger;
    fetch(`https://api.github.com/users/${userData}`)
      .then(response => response.json())
      .then(userInfo => {
        return this.setState({
          avatarUrl: userInfo.avatar_url,
          name: userInfo.name,
          location: userInfo.location,
        });
      });
  }
  render() {
    console.log(this.state)
    debugger;
    const { avatarUrl, name, location } = this.state;
    if (!avatarUrl || !name || !location) {
      return null
    }
      return (
        <div className="user">
          <img alt="User Avatar" src={avatarUrl} className="user__avatar" />
          <div className="user__info">
            <span className="user__name">{name}</span>
            <span className="user__location">{location}</span>
          </div>
        </div>
      );
  }
};

export default User;