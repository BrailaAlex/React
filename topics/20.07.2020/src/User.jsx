import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';

const User = () => {
    const [userData, setUserData] = useState(null);
    const { userId } = useParams();
    
    const fetchUserData = () => {
        console.log(userId);
        fetch(`https://api.github.com/users/${userId}`)
            .then((response) => {
                if (!response.ok) {
                    throw new Error('Oops');
                }
                return response.json();
            })
            .then((data) => {
                setUserData(data)
            })
    };

    useEffect(() => {
        fetchUserData();
    }, [userId])

    if (!userData) {
        return null;
    }

    const { name, avatar_url, location } = userData;
    return (
        <div className="user">
            <img
                alt="User Avatar"
                src={avatar_url}
                className="user__avatar"
            />
            <div className="user__info">
                <span className="user__name">{name}</span>
                <span className="user__location">{location}</span>
            </div>
        </div>
    );
    
};

export default User;