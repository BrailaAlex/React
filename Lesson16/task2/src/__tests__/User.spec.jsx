import React from 'react';
import { shallow } from 'enzyme';
import User from '../User';
import { getUserData } from '../userGateway';

jest.mock('../userGateway', () => {
  return {
    getUserData: jest.fn(() => Promise.resolve())
  }
});

describe('User', () => {
  it('should not render user if no data', () => {
    const wrappedComponent = shallow(<User />);
    expect(wrappedComponent.find('.user').exists()).toEqual(false)
  });

  it('should render user if data is received', () => {
    const wrappedComponent = shallow(<User />);
    wrappedComponent.setState({
      userData: {
        avatar_url: 'https://example.com',
        name: 'Jim',
        location: 'Planet Earth'
      }
    })
    expect(wrappedComponent.find('.user')).toMatchSnapshot();
  });
  it('should call imported function when component will be mounted', () => {
    shallow(<User />)
    expect(getUserData).toBeCalled();
  });
})