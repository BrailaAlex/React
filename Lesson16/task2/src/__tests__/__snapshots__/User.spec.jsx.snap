// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render user if data is received 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
    src="https://example.com"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      Jim
    </span>
    <span
      className="user__location"
    >
      Planet Earth
    </span>
  </div>
</div>
`;
